### PR TITLE
fix(security): Sanitize and refactor contact creation and update request

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description
This PR fix an issue where the contact forms wasn't sanitize correctly
**Fixes** # MON-6771

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [x] 2.8.x

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 1]
```diff
@@ -605,101 +479,22 @@ function updateContact_MC($contact_id = null)
...
-     */
-    if (strcmp("UPDATE contact SET ", $rq)) {
-        $rq[strlen($rq) - 2] = " ";
-        $rq .= "WHERE contact_id = '" . intval($contact_id) . "'";
-        $DBRESULT = $pearDB->query($rq);
+    $pearDB->query($rq, $bindParams);
 
-        $DBRESULT2 = $pearDB->query("SELECT contact_name FROM `contact` WHERE contact_id='" . intval($contact_id) . "' LIMIT 1");
-        $row = $DBRESULT2->fetchRow();
+    $DBRESULT2 = $pearDB->query("SELECT contact_name FROM `contact` WHERE contact_id='" . intval($contact_id) . "' LIMIT 1");
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:492]:
```suggestion
    $DBRESULT2 = $pearDB->query("SELECT contact_name FROM `contact` WHERE contact_id='" . (int)$contact_id . "' LIMIT 1");
```

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 2]
```diff
@@ -1038,3 +833,146 @@ function getContactIdByName($name)
...
+                $bindParams[$inputName] =
+                    (filter_var($inputValue, FILTER_VALIDATE_INT) === false)
+                        ? 1
+                        : (int) $inputValue;
+                break;
+            case 'contact_hostNotifOpts':
+                $bindParams['contact_host_notification_options'] =
+                    (($inputValue = filter_var(
+                        implode(",", array_keys($inputValue)),
+                        FILTER_SANITIZE_STRING
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:873]: FILTER_SANITIZE_STRING will return an empty string, not false

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 3]
```diff
@@ -1038,3 +833,146 @@ function getContactIdByName($name)
...
+                    )) === false)
+                        ? null
+                        : $inputValue;
+                break;
+            case 'contact_svNotifOpts':
+                $bindParams['contact_service_notification_options'] =
+                    (($inputValue = filter_var(
+                        implode(",", array_keys($inputValue)),
+                        FILTER_SANITIZE_STRING
+                    )) === false)
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:883]: FILTER_SANITIZE_STRING will return an empty string, not false

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 4]
```diff
@@ -1038,3 +833,146 @@ function getContactIdByName($name)
...
+                        $inputValue,
+                        $encryptType == 2 ?  'sha1' : 'md5'
+                    );
+                    $bindParams[$inputName] = $password;
+                }
+                break;
+            case 'contact_lang':
+                if (!empty($inputValue)) {
+                    $bindParams[$inputName] =
+                        (($inputValue = filter_var($inputValue, FILTER_SANITIZE_STRING)) === false)
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:941]: FILTER_SANITIZE_STRING will return an empty string, not false

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 5]
```diff
@@ -1038,3 +833,146 @@ function getContactIdByName($name)
...
+                            ? 'browser'
+                            : $inputValue;
+                }
+                break;
+            case 'contact_auth_type':
+                if (!empty($inputValue)) {
+                    $bindParams[$inputName] =
+                        (($inputValue = filter_var($inputValue, FILTER_SANITIZE_STRING)) === false)
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:949]: FILTER_SANITIZE_STRING will return an empty string, not false

File: www/include/configuration/configObject/contact/DB-Func.php [hunk 6]
```diff
@@ -1038,3 +833,146 @@ function getContactIdByName($name)
...
+                            ? 'local'
+                            : $inputValue;
+                }
+                break;
+            case 'contact_name':
+            case 'contact_alias':
```
Comment (sc979) [inline www/include/configuration/configObject/contact/DB-Func.php:955]: could these, be empty ?

Comment (sc979) [reply www/include/configuration/configObject/contact/DB-Func.php:955]: https://github.com/centreon/centreon/pull/9624/files#diff-4cfd7b0df58e097781835a57f629acf237f44234e5f7f03db093d83ba7808d24R968

**Review by sc979 — CHANGES_REQUESTED** [review]
Kindly change the check on FILTER_SANITIZE_STRING

**Review by sc979 — APPROVED** [review]

**Review by kduret — APPROVED** [review]

**Pull request merged**